src/vuln_analysis/register.py

```diff
@@ -46,27 +48,27 @@
 # pylint: enable=unused-import
 from vuln_analysis.utils.llm_engine_utils import postprocess_engine_output
 from vuln_analysis.utils.llm_engine_utils import preprocess_engine_input
-
+from nat.data_models.component_ref import FunctionRef
 logger = logging.getLogger(__name__)
 
 
 class CVEAgentWorkflowConfig(FunctionBaseConfig, name="cve_agent"):
     """
     Defines the workflow function for determining the impact of a documented CVEs on a specific project or container.
     """
-    cve_generate_vdbs_name: str = Field(description="Function name to generate vector databases")
-    cve_fetch_intel_name: str = Field(description="Function name to fetch intel")
-    cve_process_sbom_name: str = Field(description="Function name to process SBOMs")
-    cve_check_vuln_deps_name: str = Field(description="Function name to check vulnerable dependencies")
-    cve_checklist_name: str = Field(description="Function name to generate checklist")
-    cve_agent_executor_name: str = Field(description="Function name to run CVE agent on checklist")
-    cve_summarize_name: str = Field(description="Function name to generate summary")
-    cve_justify_name: str = Field(description="Function to generate justifications for each CVE")
-    cve_output_config_name: str | None = Field(default=None,
+    cve_generate_vdbs_name: FunctionRef = Field(description="Function name to generate vector databases")
+    cve_fetch_intel_name: FunctionRef = Field(description="Function name to fetch intel")
+    cve_process_sbom_name: FunctionRef = Field(description="Function name to process SBOMs")
+    cve_check_vuln_deps_name: FunctionRef = Field(description="Function name to check vulnerable dependencies")
+    cve_checklist_name: FunctionRef = Field(description="Function name to generate checklist")
+    cve_agent_executor_name: FunctionRef = Field(description="Function name to run CVE agent on checklist")
+    cve_summarize_name: FunctionRef = Field(description="Function name to generate summary")
+    cve_justify_name: FunctionRef = Field(description="Function to generate justifications for each CVE")
+    cve_output_config_name: FunctionRef | None = Field(default=None,
                                                description="Function to output workflow results "
                                                "(e.g. cve_file_output, cve_http_output). "
                                                " If None, only prints to console")
-    description: str = Field(default="Vulnerability analysis for container security workflow",
+    description: FunctionRef = Field(default="Vulnerability analysis for container security workflow",
```

Comment:
Should this still be a `str` since it's just the workflow description?
```suggestion
    description: str = Field(default="Vulnerability analysis for container security workflow",
```